csv/shared/src/main/scala/fs2/data/csv/package.scala

```diff
@@ -162,13 +220,42 @@ package object csv {
         C: CharLikeChunks[F, C],
         T: CsvRowDecoder[T, Header]): Pipe[F, C, T] = {
       if (skipHeaders)
-        lowlevel.rows(separator, quoteHandling) andThen (_.tail) andThen
+        lowlevel.rows(separator, quoteHandling) andThen lowlevel.skipHeaders andThen
```

Comment:
This doesn't actually make a difference (same implementation) though intent becomes clearer. You stated in your questions something about empty stream failing, can you elaborate? A `.tail` on an empty stream will just be an empty stream again, so no error, no problem.

Reply:
In retrospective, I think the fail had to do with the headers not being able to be decoded due to column mismatches.